src/nrtk_explorer/app/embeddings.py

```diff
@@ -165,9 +165,20 @@ def set_on_select(self, fn):
         self._on_select_fn = fn
 
     def on_select(self, indices):
-        self.state.user_selected_points_indices = indices
+        # remap transformed indices to original indices
+        original_indices = set()
+        for point_index in indices:
+            original_image_point_index = point_index
+            if point_index >= len(self.state.points_sources):
+                original_image_point_index = self.state.user_selected_points_indices[
+                    point_index - len(self.state.points_sources)
+                ]
+            original_indices.add(original_image_point_index)
+        original_indices = list(original_indices)
+
+        self.state.user_selected_points_indices = original_indices
```

Comment:
This should not be needed, in scatterplot.vue we only submit indices lower than the number of points (not transformed_points) as shown here:

```
    onSelect(indices) {
      onPanModeClick()
      indices = indices.filter((index) => index < props.points.length)
      if (scatterPlot) {
        scatterPlot.setSequences([])
      }
      emit('select', indices)
    }
```

Reply:
I changed scatterplot and added this code so we can click on transform points and the app selects them (rather than unselecting everything)

Reply:
Nice, after your last push looks better than main